cocotb/share/lib/vpi/VpiImpl.cpp

```diff
@@ -648,14 +648,19 @@ int32_t handle_vpi_callback(p_cb_data cb_data)
 
         gpi_cb_state_e new_state = cb_hdl->get_call_state();
 
-        /* We have re-primed in the handler */
-        if (new_state != GPI_PRIMED)
+        if (new_state == GPI_REPRIME) {
+            cb_hdl->set_call_state(GPI_PRIMED);
+        }
+
+        // Cleanup if the callback wasn't re-primed in the handler
+        if (new_state != GPI_PRIMED && new_state != GPI_REPRIME){
```

Comment:
`else if` or a `switch` like in the previous bits? I tend to like switches when testing enums, there are some useful warnings for catching gotchas.